modules/drivers/sqlserver/src/metabase/driver/sqlserver.clj

```diff
@@ -32,14 +33,29 @@
 (driver/register! :sqlserver, :parent :sql-jdbc)
 
 (doseq [[feature supported?] {:regex                                  false
-                              :percentile-aggregations                false
                               :case-sensitivity-string-filter-options false
                               :now                                    true
                               :datetime-diff                          true
                               :convert-timezone                       true
                               :test/jvm-timezone-setting              false}]
   (defmethod driver/database-supports? [:sqlserver feature] [_driver _feature _db] supported?))
 
+(defn- dbms-major-version
+  [db]
+  ;;;; Using try-catch as either destructuring can throw, or returned could be a non number. Happened previously with
+  ;;;; mongo. See issue #29678.
+  (try
+    (let [{[major-version] :semantic-version} (driver/dbms-version :sqlserver db)]
```

Comment:
This is expensive. Have a look at how the Mongo driver does this checks.

Reply:
I've modified the code to retrieve `:dbms_version` right from the `db` structure.